src/main/kotlin/it/pagopa/wallet/scheduler/service/RedisResumePolicyService.kt

```diff
@@ -0,0 +1,33 @@
+package it.pagopa.wallet.scheduler.service
+
+import it.pagopa.wallet.scheduler.config.properties.RedisResumePolicyConfig
+import java.time.Duration
+import java.time.Instant
+import java.util.NoSuchElementException
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+@Service
+class RedisResumePolicyService(
+    @Autowired private val redisTemplate: TimestampRedisTemplate,
+    @Autowired private val redisResumePolicyConfig: RedisResumePolicyConfig
+) : ResumePolicyService {
+    private val logger = LoggerFactory.getLogger(RedisResumePolicyService::class.java)
+
+    override fun getResumeTimestamp(target: String): Instant {
+        return redisTemplate
+            .findByKeyspaceAndTarget(redisResumePolicyConfig.keyspace, target)
+            .orElseThrow { NoSuchElementException("No resume timestamp found for target: $target") }
```

Comment:
Do not throw an exception if no checkpoint is found, it is acceptable not having it (e.g. first execution of the job); we could consider changing getResumeTimestamp return type to match the return type of findByKeyspaceAndTarget method (Optional<Instant>).

Reply:
mmm ok, i'm on it

Reply:
done here a12e3b262d3f88eb6af34fb56fc1f48869425118